src/Kiota.Builder/CodeDOM/CodeMethod.cs

```diff
@@ -15,6 +15,9 @@ public enum CodeMethodKind
     RequestGenerator,
     Serializer,
     Deserializer,
+    ComposedTypeFactory,
+    ComposedTypeDeserializer,
+    ComposedTypeSerializer,
```

Comment:
I'm a bit reluctant to introducing those new enum values. It's going to introduce inconsistencies with the other languages.
What about using the existing enum values and adding a "OriginalComposedType" property instead (nullable)? 
This way we can route things in the writer just as easily, and avoid those new values which would be confusing to new languages implementers.
Thoughts? What about @andrueastman ?

Reply:
I see what you're saying here as these are TS specific at the moment. 
We can use the `OriginalComposedType` of the return type/parameters to get the same information as they types are still factory/serialiser/deserializer while eliminating the addition of more stuff in the DOM.